runners/flink/src/main/java/org/apache/beam/runners/flink/FlinkRunner.java

```diff
@@ -74,7 +75,19 @@ protected FlinkRunner(FlinkPipelineOptions options) {
 
   @Override
   public PipelineResult run(Pipeline pipeline) {
-    SplittableParDo.convertReadBasedSplittableDoFnsToPrimitiveReadsIfNecessary(pipeline);
+    // Portable flink only support SDF as read.
```

Comment:
A side question: does this mean that portable Flink doesn't currently support "use_deprecated_read" flag? What happens if we specify "use_deprecated_read" for portable Flink? Runtime error?

Reply:
There is no `deprecated read` under `beam_fn_api` context. Splittable DoFn is the only source-like operation when with `beam_fn_api`. So if a fnapi job is using `deprecated read`, there is no runtime error but the pipeline will just stuck.

Reply:
But good part is that I don't think we have many fnapi usages on OSS runners.